packages/node/src/indexer/project.service.ts

```diff
@@ -74,7 +74,8 @@ export class ProjectService extends BaseProjectService<
     return getTimestamp(block);
   }
 
-  protected onProjectChange(project: SubqueryProject): void | Promise<void> {
+  protected async onProjectChange(project: SubqueryProject): Promise<void> {
+    await this.apiService.reloadChainTypes(project.chainTypes);
     this.apiService.updateBlockFetching();
```

Comment:
Why not combine these into a single function?

Reply:
`updateBlockFetching` also used in api service constructor, also make sense keep them independent.